### PR TITLE
Update quic-go dependency

### PR DESCRIPTION
**Pull request by fasmat**
The current version of webtransport is not compatible any more with the newest version of quic-go.

This PR updates the dependency and fixes uses of deprecated functions/types.

### CODE REVIEW

**Comment by fasmat** [comment]
@marten-seemann let me know if you want me to adjust the PR in some way 🙂

**Comment by marten-seemann** [comment]
https://github.com/quic-go/webtransport-go?tab=readme-ov-file#webtransport-go-is-currently-unfunded

File: go.mod [hunk 1]
```diff
@@ -1,12 +1,12 @@
 module github.com/quic-go/webtransport-go
 
-go 1.21
+go 1.22.0
```
Comment (marten-seemann) [inline go.mod]:
```suggestion
go 1.22
```

**Comment by fasmat** [comment]
I don't understand why `go 1.22.0` in the `go.mod` file isn't correct? `go mod tidy` is a no-op and `go build` and `go test` pass without errors.

I believe the problem is this step in the CI: https://github.com/ipdxco/unified-github-workflows/blob/61bc2a929b1a896da9559d6c4b14c574425da364/.github/workflows/go-check.yml#L36-L41

`actions/setup-go@v5` supports parsing the go file from `go.mod` directly without having to read it manually in a separate step:

```yaml
      - name: Set up the Go version read from the go.mod file
        uses: actions/setup-go@v5
        with:
          check-latest: true
          go-version-file: "release/go/go.mod"
```

Not specifying a minor version for the go directive in the `go.mod` file will cause `go mod tidy` to add one (and possibly a `toolchain` directive).

**Comment by fasmat** [comment]
However, I need to say thanks for still putting effort into this 🙂

**Comment by marten-seemann** [comment]
> Not specifying a minor version for the go directive in the `go.mod` file will sometimes cause `go mod tidy` to add one (and possibly a `toolchain` directive).

No it doesn't. See my PR, which was checked for `go mod tidy` consistency.

**Comment by marten-seemann** [comment]
> I believe the problem is this step in the CI: https://github.com/ipdxco/unified-github-workflows/blob/61bc2a929b1a896da9559d6c4b14c574425da364/.github/workflows/go-check.yml#L36-L41

If I was still maintaining this repo, I'd migrate away from these workflows. But since I'm currently not, everything will remain as is for now.

**Comment by fasmat** [comment]
> > Not specifying a minor version for the go directive in the `go.mod` file will sometimes cause `go mod tidy` to add one (and possibly a `toolchain` directive).
> 
> No it doesn't. See my PR, which was checked for `go mod tidy` consistency.

I don't know what you changed in your PR, but this failed to merge after you removed the minor version because `go mod tidy` added it back:

https://github.com/quic-go/webtransport-go/actions/runs/11396164669/job/31709549949#step:9:1

**Pull request closed without merging**